public/app/plugins/datasource/grafana-azure-monitor-datasource/resourcePicker/resourcePickerData.ts

```diff
@@ -25,12 +27,86 @@ const RESOURCE_GRAPH_URL = '/providers/Microsoft.ResourceGraph/resources?api-ver
 
 export default class ResourcePickerData extends DataSourceWithBackend<AzureMonitorQuery, AzureDataSourceJsonData> {
   private resourcePath: string;
+  private supportedMetricNamespaces: string[];
 
   constructor(instanceSettings: DataSourceInstanceSettings<AzureDataSourceJsonData>) {
     super(instanceSettings);
     this.resourcePath = `${routeNames.resourceGraph}`;
+    const cloud = getAzureCloud(instanceSettings);
+    this.supportedMetricNamespaces = new SupportedNamespaces(cloud).get();
   }
 
+  async fetchInitialRows(currentSelection: string): Promise<ResourceRowGroup> {
+    const subscriptions = await this.getSubscriptions();
+    if (!currentSelection) {
+      return subscriptions;
+    }
+
+    let resources = subscriptions;
+    const parsedURI = parseResourceURI(currentSelection);
+    if (parsedURI) {
+      const resourceGroupURI = `/subscriptions/${parsedURI.subscriptionID}/resourceGroups/${parsedURI.resourceGroup}`;
+
+      if (parsedURI.resourceGroup) {
+        const resourceGroups = await this.getResourceGroupsBySubscriptionId(parsedURI.subscriptionID);
+        resources = addResources(resources, `/subscriptions/${parsedURI.subscriptionID}`, resourceGroups);
+      }
+
+      if (parsedURI.resource) {
+        const resourcesForResourceGroup = await this.getResourcesForResourceGroup(resourceGroupURI);
+        resources = addResources(resources, resourceGroupURI, resourcesForResourceGroup);
+      }
+    }
+    return resources;
+  }
+
+  async fetchNestedRowData(resourceGroupOrSubscription: ResourceRow) {
+    return resourceGroupOrSubscription.type === ResourceRowType.Subscription
+      ? await this.getResourceGroupsBySubscriptionId(resourceGroupOrSubscription.id)
+      : await this.getResourcesForResourceGroup(resourceGroupOrSubscription.id);
+  }
+
+  search = async (searchPhrase: string, selectableEntryTypes: ResourceRowType[]): Promise<ResourceRowGroup> => {
+    let query;
+    if (selectableEntryTypes.length === 1 && selectableEntryTypes[0] === ResourceRowType.Resource) {
+      query = `
+      resources
+      | where id contains "${searchPhrase}"
+      | where type in (${this.supportedMetricNamespaces.map((ns) => `"${ns.toLowerCase()}"`).join(',')})
+      | order by tolower(name) asc
+      | limit 200
```

Comment:
curious, why the limit to 200?

Reply:
I believe this is what Azure Portal does, it seemed like a reasonable number to start with?